kong/plugins/jwt/handler.lua

```diff
@@ -38,20 +38,25 @@ local function retrieve_token(request, conf)
     end
   end
 
-  local authorization_header = request.get_headers()["authorization"]
-  if authorization_header then
-    local iterator, iter_err = ngx_re_gmatch(authorization_header, "\\s*[Bb]earer\\s+(.+)")
-    if not iterator then
-      return nil, iter_err
-    end
+  local request_headers = request.get_headers()
+  for _, v in ipairs(conf.header_names) do
+    local token_header = request_headers[v]
+    if token_header then
```

Comment:
this value `token_header` can potentially not be a string, but a table. This happens when the same header occurs multiple times.

Add something like:
```lua
if type(token_header) == "table" then
  token_header = token_header[1]
end
```

would be nice to add a regression test for this as well (not failing on duplicate headers)

Reply:
@Tieske, sorry for the delay. I add the test and fix the case with a table instead a string. Should I reopen the merge request and close this one?